control-plane/agents/src/bin/core/volume/operations_helper.rs

```diff
@@ -554,6 +554,23 @@ impl OperationGuardArc<VolumeSpec> {
         Ok(())
     }
 
+    /// Resize the volume target(nexus).
+    pub(super) async fn resize_volume_target(
+        &self,
+        registry: &Registry,
+        nexus_grd: &mut OperationGuard<ResourceMutex<NexusSpec>, NexusSpec>,
+        requested_size: u64,
+    ) -> Result<(), SvcError> {
+        let nexus = registry.nexus(nexus_grd.as_ref().uid()).await?;
```

Comment:
Why is this needed? nexux_grd already has node and uuid?

Reply:
Yeah, I think earlier this function was doing something more that needed state, but is now simplified and this remained. Using from nexus_grd now but need to clone, otherwise it complains of immutable borrow attempt 
if directly passed in as `&nexus_grd.lock().node`
```
        let node = nexus_grd.lock().node.clone();
        let nexus_uuid = nexus_grd.uuid().clone();
        nexus_grd
            .resize(
                registry,
                &ResizeNexus::new(&node, &nexus_uuid, requested_size),
            )
            .await?;

```

Reply:
Can use nexus_grd.as_ref().node.clone();